tests/functional/test_show.py

```diff
@@ -78,6 +78,26 @@ def test_find_package_not_found():
     result = search_packages_info(['abcd3'])
     assert len(list(result)) == 0
 
+def test_report_single_not_found(script):
+    """
+    Test that a single unmatched package is reported not found.
+    """
+    result = script.pip('show', 'abcd3', expect_error=True)
```

Comment:
You'll want `allow_stderr_warning`, as this will assert a non-zero exit code. Likewise below.

Reply:
@chrahunt : Eventually  figured the same out, _then spotted your message_!

Reply:
To clarify, by “this” he means “expect_error” asserts a non-zero exit code.